cpp/src/groupby/common/kernel_utils.hpp

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _GROUPBY_KERNEL_UTILS_H
+#define _GROUPBY_KERNEL_UTILS_H
+
+namespace cudf {
+namespace groupby {
+
+/**---------------------------------------------------------------------------*
+ * @brief Base case for invalid SourceType and op combinations.
```

Comment:
The brief should say what the functor does, not just what the template specialization is.